application/client/src/app/ui/views/toolbar/chart/chart/chart.ts

```diff
@@ -0,0 +1,411 @@
+            this._updateHasNoData();
+            return;
+        }
+        this._labelState.loading = true;
+        this._chart.data.labels = [];
+        for (const [line, values] of this._session.search.values.get()) {
+            values.forEach((sValue: string, id: number) => {
+                const value: number = parseInt(sValue);
+                const scaleX = (this._chart.config.options as any).scales['x'];
+                if (line < scaleX.min) {
+                    scaleX.min = line;
+                } else if (line > scaleX.max) {
+                    scaleX.max = line;
+                }
+                (this._chart.data.labels as number[]).push(line);
+                this._chart.data.datasets[id].data.push({ x: line, y: value });
+            });
+            this._chart.data.labels = [...new Set(this._chart.data.labels as number[])].sort(
+                (a, b) => a - b,
+            );
+        }
+        this._labelState.loading = false;
+        this._chart.update();
+        this._updateHasNoData();
+    }
+
+    // Postponed due to overlapping with filters canvas
+    private _showYAxis(request: ChartRequest) {
+        const uuid: string = request.uuid();
+        this._chart.update();
+        if (
+            this._chart.options.scales !== undefined &&
+            this._chart.options.scales[uuid] !== undefined
+        ) {
+            const yAxis = this._chart.options.scales[uuid] as ScaleOptionsByType<
+                'radialLinear' | keyof CartesianScaleTypeRegistry
+            >;
+            yAxis.display = this._session.search.store().charts().selectedGuid === uuid;
+            yAxis.ticks.color = request.definition.color;
+        }
+    }
+
+    // Postponed due to overlapping with filters canvas
+    private _updateYAxes(entities: ChartRequest[]) {
+        this._chart.update();
+        const selectedUuid: string = this._session.search.store().charts().selectedGuid;
+        Object.keys(
+            this._chart.options.scales as {
+                [key: string]: ScaleOptionsByType<
+                    'radialLinear' | keyof CartesianScaleTypeRegistry
+                >;
+            },
+        ).forEach((axis: string) => {
+            if (axis === 'x' || axis === 'y') {
+                return;
+            }
+            const entity: ChartRequest | undefined = entities.find(
+                (entity: ChartRequest) => entity.uuid() === axis,
+            );
+            if (entity === undefined) {
+                delete (this._chart.options as any).scales[axis];
+            } else {
+                if (axis === selectedUuid && entity.definition.active) {
+                    (this._chart.options as any).scales[axis].display = true;
+                    (this._chart.options as any).scales[axis].color;
+                }
+                (this._chart.options as any).scales[axis].display =
+                    axis === selectedUuid && entity.definition.active;
```

Comment:
so many `any` :((

Reply:
Well just for line 260 we'd have to write like this:
```
                    const options = this._chart.options;
                    if (options !== undefined) {
                        const scales = options.scales;
                        if (scales !== undefined) {
                            const scale = scales[axis];
                            if (scale !== undefined) {
                                scale.display = true;
                            }
                        }
```